### DISCUSSION TITLE
GET layoutComponents return 404

### DISCUSSION THREAD

**Discussion by mrdrivingduck** [vuepress/core#1173] [Q&A] [answered]
Recently, I have upgraded my project from 2.0.0-beta-48 to 53. When I tried to preview my website locally, I got:

```
GET http://localhost:8080/${project_name}/home/${user_name}/${project_name}/docs/.vuepress/.temp/internal/layoutComponents net::ERR_ABORTED 404 (Not Found)
```

Seems it is a invalid path.

But when I comment the docSearch plugin in my config, the error no longer exists:

```ts
  plugins: [
    // docsearchPlugin({
    //   appId: "xxx",
    //   apiKey: "xxx",
    //   indexName: "xxx",
    //   locales: {
    //     "/": {
    //       placeholder: "Search Documentation",
    //     },
    //   },
    // }),
  ]
```

Any solutions by the maintainers? Thanks.

**Comment by Mister-Hope** [comment, marked as answer]
Check whether you are using same version of vuepress and all it's subpackages and official plugins.

**Reply by mrdrivingduck** [reply]
Yes. They are all 53 version, including vuepress-vite.

**Reply by Mister-Hope** [reply]
Then you should provide a repro, also make sure you are no using beta.50-pre.x

**Reply by mrdrivingduck** [reply]
https://github.com/mrdrivingduck/vp-test

Here is a minimal reproduce. Comment/uncomment docSearch plugin in `config.ts` to reveal the problem.

**Reply by Mister-Hope** [reply]
This is obviously an invalid repro

![Screenshot_2022-11-28-16-36-50-64_320a9a695de7cdce83ed5281148d6f19.jpg](https://user-images.githubusercontent.com/33315834/204231411-9fdd1920-1216-4b1c-ba3f-9240d86f0089.jpg)

I believe there are multiple version of vuepress existing in your project

**Reply by mrdrivingduck** [reply]
By the time the dependencies are as small as following, the issue still exists.

```json
"devDependencies": {
  "@vuepress/plugin-docsearch": "^2.0.0-beta.53",
  "prettier": "^2.7.1",
  "vuepress": "^2.0.0-beta.53"
}
```

They are all VuePress official component right?

**Reply by Mister-Hope** [reply]
Yes, but with another example at https://github.com/vuepress-theme-hope/vuepress-theme-hope

There are 17 documentation sites using same deps above, but I am not having any issues, that's why I am expecting a valid repro

**Reply by Mister-Hope** [reply]
If possiable, provide one with https://v2.vuepress.vuejs.org/new

**Reply by mrdrivingduck** [reply]
The website https://v2.vuepress.vuejs.org/new keeps running on `Running Start Command` for a long time:

![image](https://user-images.githubusercontent.com/32560442/204588344-df1a003f-9310-4a0c-8eb9-234a4108670a.png)

BTW, what do you mean for a **valid** repro? I think the dependencies in `package.json` of my project is different from vuepress-theme-hope repo, at least you are not using `"vuepress": "^2.0.0-beta.53"`. I have tried to minimize the issue in [this repo](https://github.com/mrdrivingduck/vp-test), just clone it and `yarn` is enough for reproduce.

**Reply by mrdrivingduck** [reply]
@Mister-Hope Since there is no update, is it an issue of #1195 ? Does yarn has a similar problem?

**Reply by Mister-Hope** [reply]
If you stuck at stackblitz, check the console to see if you run out of memory. A browser worker can no t request system to provide more memory for itself if the provided memory is not enough

**Reply by Mister-Hope** [reply]
![image](https://user-images.githubusercontent.com/33315834/207545251-3b26f8a1-55d5-4db8-a337-c5fca1141d8c.png)

> Check whether you are using same version of vuepress and all it's subpackages and official plugins.

I think I already told you enough content about why this is "invalid". Also we already have an issue pined at out repo which describe clearly the solution. So what else can I do for you? (Sign)

**Reply by mrdrivingduck** [reply]
Why do you express an attitude of sigh (if I am not understand mistakenly by the word `sign`)?

As far as I can see:

1. I have done what you have told me to do, trying best to prove the problem is not on my side
2. Half a month later from my question, more and more people are meeting up with similar problem, and turns out it is a problem of npm, the so called invalid repro is not something I can control
3. the purpose I asked the question is to make sure the problem has been solved, is it a rude enough question to make you sigh?

**Reply by Mister-Hope** [reply]
1. I am a Collaborator of the repo, and I have no rights to publish package, so I can not contact npm support. Meteorlxy should be in charged of those things.
2. This is an open source project and I have study and work to do, also I gets 50+ emails and 30+ github notification per day, so I don't think  I shall ensure I reply all of them. Issue is a place I will check in periods, but not dicussion, as discussion probably means that this is a help or other thing personal
3. For the late reply, I believed I ahead to stackbliuz to check what you are reporting (I have memory on this, but there must be something which stops me from finish the reproduction and reply you like study, I do be busy)
4. I still think that I provide explaination and solution to this, as I pointed out in the second reply "Then you should provide a repro, also make sure you are no using beta.50-pre.x" and I still get a repo with `beta.50-pre.x`. It's not about your attitude, it's just me a poor postgraduate student majored in theoretical physics busy at study all day, and still find a notification which ask me to have a little more explain politely. 
>  Does yarn has a similar problem?

Yes, but I absolutely do not want to answer question like this, because it's definitely out of scope for me to explain how node package version works and what's a  package manager and "will I get different pacakge version with differernt pacakge manager". The problem is about the first one, and definitely nothing related to 2 so the 3 is "NO".

If my reply confuse you, I would like to apologize, but I am sure you know that a lot of open source repo maintainers won't check discussion (e.g: vuejs/core), so at least you should somehow agree that it's not a fault for a maintainer to leave a discussion here just because he forgot it.

**Reply by meteorlxy** [reply]
I'm still talking with npm support about those fxxking pre-x versions. 😃

**Reply by mrdrivingduck** [reply]
@Mister-Hope I'm not complaining about the late reply from you. I pretty understand most open source project maintainers work as part-time. Also, I'm not expecting any reply from you, which is the reason I asked the question in discussion part. I just want to try my luck if anyone can help, because my work is mostly about C programming, so I'm not familiar with front end techs.

I am just confused about the word `sign` because it seems that you are inpatient with this kind of question. It is okay if you ignore my question. And also, thanks for your detailed explanation of my question, it really helps. Wish you good to your study.

**Reply by Mister-Hope** [reply]
> I'm still talking with npm support about those fxxking pre-x versions. 😃

So any actual progress?
